crates/router/src/core/payment_methods/network_tokenization.rs

```diff
@@ -778,24 +778,125 @@ pub async fn check_token_status_with_tokenization_service(
+        services::Method::Post,
+        tokenization_service.check_token_status_url.as_str(),
+    );
+    let payload = pm_types::CheckTokenStatus {
+        card_reference: network_token_requestor_reference_id,
+        customer_id: customer_id.clone(),
+    };
+
+    request.add_header(headers::CONTENT_TYPE, "application/json".into());
+    request.add_header(
+        headers::AUTHORIZATION,
+        tokenization_service
+            .token_service_api_key
+            .clone()
+            .peek()
+            .clone()
+            .into_masked(),
+    );
+    request.add_default_headers();
+    request.set_body(RequestContent::Json(Box::new(payload)));
+
+    // Send the request using `call_connector_api`
+    let response = services::call_connector_api(state, request, "Check Network token Status")
+        .await
+        .change_context(errors::NetworkTokenizationError::ApiError);
+    let res = response
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+        .attach_printable("Error while receiving response")
+        .and_then(|inner| match inner {
+            Err(err_res) => {
+                let parsed_error: pm_types::NetworkTokenErrorResponse = err_res
+                    .response
+                    .parse_struct("Network Tokenization Error Response")
+                    .change_context(
+                        errors::NetworkTokenizationError::ResponseDeserializationFailed,
+                    )?;
+                logger::error!(
+                    error_code = %parsed_error.error_info.code,
+                    developer_message = %parsed_error.error_info.developer_message,
+                    "Network tokenization error: {}",
+                    parsed_error.error_message
+                );
+                Err(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+                    .attach_printable(format!("Response Deserialization Failed: {err_res:?}"))
+            }
+            Ok(res) => Ok(res),
+        })
+        .inspect_err(|err| {
+            logger::error!("Error while deserializing response: {:?}", err);
+        })?;
+
+    let check_token_status_response: pm_types::CheckTokenStatusResponse = res
+        .response
+        .parse_struct("Network Token Status Response")
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)?;
+
+    Ok(check_token_status_response)
+}
+
+#[cfg(feature = "v2")]
+pub async fn do_status_check_for_network_token(
+    state: &routes::SessionState,
+    payment_method_info: &domain::PaymentMethod,
+) -> CustomResult<pm_types::CheckTokenStatusResponse, errors::NetworkTokenizationError> {
+    let network_token_requestor_reference_id = payment_method_info
+        .network_token_requestor_reference_id
+        .clone();
+
+    if let Some(ref_id) = network_token_requestor_reference_id {
+        if let Some(network_tokenization_service) = &state.conf.network_tokenization_service {
```

Comment:
make use of match here
```suggestion
    match (network_token_requestor_reference_id, &state.conf.network_tokenization_service ){
        (Some(ref_id), Some(network_tokenization_service)) {
```